app/scripts/controllers/detect-tokens.js

```diff
@@ -50,14 +49,15 @@ export default class DetectTokensController {
     this.network = network;
     this.keyringMemStore = keyringMemStore;
     this.tokenList = tokenList;
+    this.useTokenDetection =
+      this.preferences?.store.getState().useTokenDetection;
     this.selectedAddress = this.preferences?.store.getState().selectedAddress;
     this.tokenAddresses = this.tokensController?.state.tokens.map((token) => {
       return token.address;
     });
     this.hiddenTokens = this.tokensController?.state.ignoredTokens;
-    this.detectedTokens = process.env.TOKEN_DETECTION_V2
-      ? this.tokensController?.state.detectedTokens
-      : [];
+    this.detectedTokens = this.tokensController?.state.detectedTokens;
+    this.chainId = network?.store.getState().provider.chainId;
```

Comment:
I see there is an optional chaining operator on `network` here. Should we also have it on all references to `this._network`?

Reply:
There are multiple places in this file where we try to get chainId by doing this `network?.store.getState().provider.chainId`

I'd suggest adding a private method, or a utility function:

```
function getChainIdFromNetworkController(network) {
    return network?.store?.getState().provider?.chainId
}
```
And use that throughout the file